### PR TITLE
fix(logging): blacklist endpoints with sensitive data from logging body

### PR DESCRIPTION
**Pull request by jsteenb2**
blacklists endpoints with sensitive data from having the body logged

- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass

### CODE REVIEW

File: http/middleware.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+		return http.HandlerFunc(fn)
+	}
+}
+
+type isValidMethodFn func(method string) bool
+
+var (
+	ignoreMethod = func(string) bool { return true }
+
+	blacklistEndpoints = map[string]isValidMethodFn{
```
Comment (jademcgough) [inline http/middleware.go]: Nice pattern 👍

Can we also add write and query endpoints to the blacklist? They get hit a lot, and we really don't need to log all those requests.

Comment (jsteenb2) [reply http/middleware.go]: great point, I'll definitely add that in 👍

Comment (jsteenb2) [reply http/middleware.go]: @jademcgough pushed up the changes 👍

Comment (jsteenb2) [reply http/middleware.go]: @jademcgough I went through more of the request and found a few more that would have secrets. The org endpoints I totally missed.

Comment (jademcgough) [reply http/middleware.go]: @jsteenb2 ah right, I was pretty sure there was something I was forgetting yesterday, it was the org secrets path :)

**Comment by jsteenb2** [comment]
@jademcgough totally agree. I started down that route to do that but it was getting a bit hacky matching on strings. I ended up punting on it b/c the tests were super brittle. I went back and added it back and I'm glad I did! I had missed the `:` prefixes for wildcard matches which would have have all those endpoints body printed. I know have tests for the entire blacklist as well as some base level stuff.  Thanks again for beating the test drum 🤘 !

**Pull request merged**